### PR TITLE
utilities don't handle dataWindows more than 65,535 wide/high unless '-x' arg is provided

### PR DESCRIPTION
**Pull request by peterhillman**
Prevents excessive memory allocation errors in examples in #245
This change will mean scripts processing large images will need to be modified, but the library itself isn't changed. `-x` was the most convenient flag that was unused by all utilities.

### CODE REVIEW

File: OpenEXR/exr2aces/main.cpp [hunk 1]
```diff
@@ -87,6 +87,9 @@ usageMessage (const char argv0[], bool verbose = false)
 		"\n"
 		"Options:\n"
 		"\n"
+        "-x        support large images: remove 65535 pixel limit on image\n"
+        "          width/height (requires significant memory to process)\n"
+		"\n"
```
Comment (kdt3rd) [inline OpenEXR/exr2aces/main.cpp:93]: Looks like you've got the odd space vs. tab issue. We really need to convert the whole repo to clang-format or equivalent.

File: OpenEXR/exrenvmap/main.cpp [hunk 2]
```diff
@@ -143,13 +143,16 @@ usageMessage (const char argv0[], bool verbose = false)
                 "-d         sets level size rounding to ROUND_DOWN (default)\n"
                 "\n"
                 "-u         sets level size rounding to ROUND_UP\n"
-                "\n"
+                "\n"
```
Comment (kdt3rd) [inline OpenEXR/exrenvmap/main.cpp]: extra spaces at EOL

File: OpenEXR/exrmakepreview/main.cpp [hunk 3]
```diff
@@ -71,6 +71,9 @@ usageMessage (const char argv0[], bool verbose = false)
 		"-e s      adjusts the preview image's exposure by s f-stops\n"
 		"          (default is 0).  Positive values make the image\n"
 		"          brighter, negative values make it darker.\n"
+        "\n"
+        "-x        support large images: remove 65535 pixel limit on image\n"
+        "          width/height (requires significant memory to process)\n"
 		"\n"
```
Comment (kdt3rd) [inline OpenEXR/exrmakepreview/main.cpp:77]: space vs. tab

File: OpenEXR/exrmakepreview/main.cpp [hunk 4]
```diff
@@ -182,7 +193,7 @@ main(int argc, char **argv)
 
     try
     {
-	makePreview (inFile, outFile, previewWidth, exposure, verbose);
+	makePreview (inFile, outFile, maximumSize , previewWidth, exposure, verbose);
```
Comment (kdt3rd) [inline OpenEXR/exrmakepreview/main.cpp]: extra space

File: OpenEXR/exrmakepreview/makePreview.cpp [hunk 5]
```diff
@@ -145,10 +145,15 @@ generatePreview (const char inFileName[],
 void
 makePreview (const char inFileName[],
 	     const char outFileName[],
+         int maximumSize,
```
Comment (kdt3rd) [inline OpenEXR/exrmakepreview/makePreview.cpp]: space vs tab

File: OpenEXR/exrmakepreview/makePreview.h [hunk 6]
```diff
@@ -44,6 +44,7 @@
 
 void	makePreview (const char inFileName[],
 	             const char outFileName[],
+             int maximumSize,
```
Comment (kdt3rd) [inline OpenEXR/exrmakepreview/makePreview.h:47]: space vs tab

File: OpenEXR/exrmaketiled/main.cpp [hunk 7]
```diff
@@ -98,8 +98,11 @@ usageMessage (const char argv0[], bool verbose = false)
         "\n"
         "-z x      sets the data compression method to x\n"
         "          (none/rle/zip/piz/pxr24/b44/b44a/dwaa/dwab,\n"
-        "          default is zip)\n"
+        "          default is zip)\n"        
         "\n"
+        "-x        support large images: remove 65535 pixel limit on image\n"
+        "          width/height (requires significant memory to process)\n"
+		"\n"
```
Comment (kdt3rd) [inline OpenEXR/exrmaketiled/main.cpp:105]: space vs tab

File: OpenEXR/exrmultiview/main.cpp [hunk 8]
```diff
@@ -86,8 +86,9 @@ usageMessage (const char argv0[], bool verbose = false)
 		"\n"
 		"-v        verbose mode\n"
 		"\n"
-		"-h        prints this message\n";
-
+		"-h        prints this message\n"
+        " -x         support large images: remove 65535 pixel limit on image\n"
+        "            width/height (requires significant memory to process)\n";
 	 cerr << endl;
```
Comment (kdt3rd) [inline OpenEXR/exrmultiview/main.cpp:92]: space vs tab

File: OpenEXR/exrmultiview/main.cpp [hunk 9]
```diff
@@ -180,6 +182,11 @@ main(int argc, char **argv)
 	    compression = getCompression (argv[i + 1]);
 	    i += 2;
 	}
+	else if(!strcmp (argv[i],"-x"))
+    {
+        maximumSize = 0;
+        i += 1;
+    }
 	else if (!strcmp (argv[i], "-v"))
```
Comment (kdt3rd) [inline OpenEXR/exrmultiview/main.cpp:190]: space vs tab

File: OpenEXR/exrmultiview/makeMultiView.cpp [hunk 10]
```diff
@@ -64,9 +64,12 @@ void
 makeMultiView (const vector <string> &viewNames,
 	       const vector <const char *> &inFileNames,
 	       const char *outFileName,
+           int maximumSize,
```
Comment (kdt3rd) [inline OpenEXR/exrmultiview/makeMultiView.cpp]: space vs tab

File: OpenEXR/exrmultiview/makeMultiView.h [hunk 11]
```diff
@@ -52,6 +52,7 @@
 void	makeMultiView (const std::vector <std::string> &viewNames,
 		       const std::vector <const char *> &inFileNames,
 		       const char *outFileName,
+               int maximumSize,
```
Comment (kdt3rd) [inline OpenEXR/exrmultiview/makeMultiView.h]: space vs tab

File: OpenEXR/exrstdattr/main.cpp [hunk 12]
```diff
@@ -234,7 +234,11 @@ usageMessage (const char argv0[], bool verbose = false)
 		"\n"
 		"Other Commands:\n"
 		"\n"
-		"  -h        prints this message\n";
+		" -h         prints this message\n"
+                "\n"
+        " -x         support large images: remove 65535 pixel limit on image\n"
+        "            width/height (requires significant memory to process)\n";
```
Comment (kdt3rd) [inline OpenEXR/exrstdattr/main.cpp:240]: space vs tab

File: OpenEXR/exrstdattr/main.cpp [hunk 13]
```diff
@@ -801,6 +807,12 @@ main(int argc, char **argv)
 	    {
 		usageMessage (argv[0], true);
 	    }
+	    else if(!strcmp(argv[i],"-x"))
+        {
+            Header::setMaxImageSize(0,0);
+            Header::setMaxTileSize(0,0);
+            i+=1;
+        }
 	    else
```
Comment (kdt3rd) [inline OpenEXR/exrstdattr/main.cpp:816]: space vs tab

**Review by kdt3rd — APPROVED** [review]
I tagged the places it looks like there's some whitespace nigglies. Otherwise, looks great

**Comment by cary-ilm** [comment]
The TSC discussed this and concluded it's an unnecessary feature, not worth the added complexity. Closing the issues for now. If the issues is raised again in the future, we can revisit.

**Pull request closed without merging**